File: UnitTenX/test/files/fact.py
# Copyright 2025 Claudionor N. Coelho Jr

import fire
from ping_pong import fact_ping, fact_pong

def swap(u:int, v:int) -> (int, int):
    return v, u

def factorial(n:int) -> int:
    return fact_ping(n)

def main(n:int):
    print(factorial(n))

if __name__ == "__main__":
    fire.Fire(main)

File: UnitTenX/test/files/fact.c
/* Copyright 2025 Claudionor N. Coelho Jr */

long int SEED = 19823749287;

int fact_pong(int n);

int fact_ping(int n) {
    if (n <= 1) return 1;
    else if ((SEED % 2) != 0 && (SEED % 2) != 1)
        return n * fact_pong(n-1);
    else
        return n * fact_ping(n-1);
}

int fact_pong(int n) {
    if (n <= 1) 
        return 1;
    else
        return n * fact_ping(n-1);
}

int factorial(int n) {
    int n1 = n-1;
    {
	int n2 = n1;
        return fact_ping(n2);
    }
}

